packages/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -637,10 +657,10 @@ export const useGridRows = (
       }
     }
 
-    logger.debug(`Updating all rows, new length ${props.rows.length}`);
+    logger.debug(`Updating all rows, new length ${props.rows?.length}`);
```

Comment:
Sidenote but these logger calls have the disadvantage that we need to create a new string in memory every time it runs, even if it doesn't log anything. If it's in a hot path, it could affect performance. A better option would be printf style calls that don't allocate new strings in the arguments, e.g. `printf('new length: %i', props.rows.length)`.